src/mcpadapt/smolagents_adapter.py

```diff
@@ -134,34 +249,23 @@ def forward(
                     else:
                         import torchaudio  # type: ignore
 
-                        audio_data = base64.b64decode(content.data)
+                        audio_data = base64.b64decode(content_item.data)
                         audio_io = BytesIO(audio_data)
                         audio_tensor, _ = torchaudio.load(audio_io)
                         return audio_tensor
 
-                raise ValueError(
-                    f"tool {self.name} returned an unsupported content type: {type(content)}"
-                )
-
-        # make sure jsonref are resolved
-        input_schema = {
-            k: v
-            for k, v in jsonref.replace_refs(mcp_tool.inputSchema).items()
-            if k != "$defs"
-        }
-
-        # make sure mandatory `description` and `type` is provided for each arguments:
-        for k, v in input_schema["properties"].items():
-            if "description" not in v:
-                input_schema["properties"][k]["description"] = "see tool description"
-            if "type" not in v:
-                input_schema["properties"][k]["type"] = "string"
+                else:
+                    raise ValueError(
+                        f"tool {self.name} returned an unsupported content type: {type(content_item)}"
+                    )
 
         tool = MCPAdaptTool(
             name=mcp_tool.name,
             description=mcp_tool.description or "",
             inputs=input_schema["properties"],
-            output_type="object",
+            output_type=output_type,
+            output_schema=output_schema,
+            use_structured_features=self.structured_output,
```

Comment:
I think for consitency it would make more sense to rename use_strcutred_features to structured_output as it is on the adapter class?